pkg/virtctl/expose/expose_test.go

```diff
@@ -2,367 +2,381 @@ package expose_test
+			Expect(err).ToNot(HaveOccurred())
+			err = runCommand("vmirs", vmirs.Name, "--name", "my-service")
+			Expect(err).To(MatchError(ContainSubstring("cannot expose VirtualMachineInstanceReplicaSet with match expressions")))
 		})
-		Context("With node-port service on a vm", func() {
-			It("should succeed", func() {
-				cmd := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vmi", vmName, "--name", "my-service",
-					"--port", "9999", "--type", "NodePort")
-				Expect(cmd()).To(Succeed())
+	})
+
+	Context("should succeed", func() {
+		const (
+			labelKey   = "my-key"
+			labelValue = "my-value"
+
+			serviceName    = "my-service"
+			servicePort    = int32(9999)
+			servicePortStr = "9999"
+		)
+
+		var (
+			vmi            *v1.VirtualMachineInstance
+			vm             *v1.VirtualMachine
+			vmirs          *v1.VirtualMachineInstanceReplicaSet
+			createdService *k8sv1.Service
+			resources      map[string]string
+		)
+
+		BeforeEach(func() {
+			kubeClient.Fake.PrependReactor("create", "services", func(action testing.Action) (bool, runtime.Object, error) {
+				createAction, ok := action.(testing.CreateAction)
+				Expect(ok).To(BeTrue())
+				createdService, ok = createAction.GetObject().(*k8sv1.Service)
+				Expect(ok).To(BeTrue())
+
+				// These assertions should be true for all created services
+				Expect(createdService.Name).To(Equal(serviceName))
+				Expect(createdService.Spec.Selector).To(HaveLen(1))
+				Expect(createdService.Spec.Selector).To(HaveKeyWithValue(labelKey, labelValue))
+
+				return true, createdService, nil
 			})
+
+			vmi = libvmi.New(libvmi.WithLabel(labelKey, labelValue))
+			vmi, err := virtClient.KubevirtV1().VirtualMachineInstances(metav1.NamespaceDefault).Create(context.Background(), vmi, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			vm, err = virtClient.KubevirtV1().VirtualMachines(metav1.NamespaceDefault).Create(context.Background(), libvmi.NewVirtualMachine(vmi), metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			vmirs = kubecli.NewMinimalVirtualMachineInstanceReplicaSet("vmirs")
+			vmirs.Spec.Template = &v1.VirtualMachineInstanceTemplateSpec{}
+			vmirs.Spec.Selector = &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					labelKey: labelValue,
+				},
+			}
+			vmirs, err = virtClient.KubevirtV1().VirtualMachineInstanceReplicaSets(metav1.NamespaceDefault).Create(context.Background(), vmirs, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			resources = map[string]string{
+				"vmi":   vmi.Name,
+				"vm":    vm.Name,
+				"vmirs": vmirs.Name,
+			}
 		})
-		Context("With cluster-ip on an vm", func() {
-			It("should succeed", func() {
-				err := clientcmd.NewRepeatableVirtctlCommand(expose.COMMAND_EXPOSE, "vm", vmName, "--name", "my-service",
-					"--port", "9999")
-				Expect(err()).To(Succeed())
-			})
+
+		It("creating a service with default settings", func() {
+			for resourceType, resourceName := range resources {
```

Comment:
I think you should convert it to a table test and avoid grouping this in one test.
If there is a problem, it will be harder to understand on which resource the problem is and it will not run the other resources.

This is relevant to all the other tests below.

Reply:
Right, addressed.